tests/composite_pk/test_filter.py

```diff
@@ -607,3 +607,31 @@ def setUp(self):
         )
         self.enterContext(feature_patch_1)
         self.enterContext(feature_patch_2)
+
+
+class CompositePKExcludeNoneTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.tenant = Tenant.objects.create()
+        cls.user = User.objects.create(
+            tenant=cls.tenant, id=1, email="exclude@example.com"
+        )
+        cls.user_pk = cls.user.pk
+        cls.partial_none = (cls.user_pk[0], None)
+        cls.full_none = (None, None)
```

Comment:
IMO this decreases readability. I would just inline the values in the tests so it's not necessary to find where they're defined.